File: Data Science and Machine Learning Essentials/Lab2C Data Quantization in Azure ML/DAT203x_Lab02C/Quantize.R
frame1 <- maml.mapInputPort(1)
## Bin the wind column into 4 categories.
bins <- c(0,   2.5,   5, 7.5, 10)
frame1[, "wind_cat"] <- cut(frame1[, "wind"], breaks = bins)

## Create categorical variables from the X and Y columns.
frame1[, c("X", "Y")] <- lapply(frame1[, c("X", "Y")], as.factor)

## Output the data frame.
maml.mapOutputPort('frame1')

File: Data Science and Machine Learning Essentials/Lab2C Data Quantization in Azure ML/DAT203x_Lab02C/Quantize.py
def azureml_main(frame1):
## Quantize the wind into 4 categories using cut with explicit columns.
  import pandas as pd
  bins = [0, 2.5, 5, 7.5, 10] 
  frame1['wind_cat'] = pd.cut(frame1['wind'], bins)
  return frame1
